### ISSUE TITLE
A simple labelling tool written in Python

### ISSUE THREAD

**Issue by anuragxel** [facebookresearch/segment-anything#128]
Wrote a labelling tool leveraging SAM's ONNX decoder. Takes a folder of images and their embeddings, lets a user annotate objects in a simple interface and saves the annotations in COCO format. Suggestions and Contributions appreciated.

https://github.com/anuragxel/salt

**Comment by prathamjagga** [comment]
That is awesome

**Comment by vietanhdev** [comment]
Check out my work of **Segment Anything** integration to [AnyLabeling](https://github.com/vietanhdev/anylabeling) - a fork of **LabelMe**.
Video: <https://www.youtube.com/watch?v=5iQSGL7ebXE>
**Note:** Work in progress. Welcome your comments.
![image](https://user-images.githubusercontent.com/18329471/231259405-0858af91-527c-4ad3-bfc6-fa64b62f2649.png)

**Comment by ryouchinsa** [comment]
Here is the [Segment Anything CPP Wrapper for macOS](https://github.com/ryouchinsa/sam-cpp-macos/tree/master).
This code is originated from [Segment Anything CPP Wrapper](https://github.com/dinglufe/segment-anything-cpp-wrapper) and implemented on macOS app [RectLabel](https://rectlabel.com). We customized the original code so that getMask() uses the previous mask result called as low_res_logits and retain the previous mask array for undo/redo actions. 

We hope this macOS version would help to develop the iOS version.
Please let us know your opinion.

![sam_polygon](https://github.com/ryouchinsa/sam-cpp-macos/assets/1954306/4640e139-c533-4b8c-b27b-e02a401b9bbd)

**Comment by ryouchinsa** [comment]
Improved "Create polygon using SAM" feature so that you can label pixels using the pixels option. You can label [Segment Anything 1 Billion (SA-1B)](https://ai.meta.com/datasets/segment-anything-downloads/) like dataset by yourself.

![sam-pixels](https://github.com/facebookresearch/segment-anything/assets/1954306/d828c18b-6fda-4258-b866-21bc34a0d17f)
